docstore/doc.go

```diff
@@ -12,42 +12,42 @@
-// A document store is a service that stores data in recursively defined
-// documents (like, but not necessarily the same as, JSON). Like other
-// NoSQL databases, document stores are schemaless.
+// Package docstore provides a portable way of interacting with a document store.
+// A document store is a service that stores data in semi-structured JSON-like
+// documents. Like other NoSQL databases, document stores are schemaless.
 // See https://en.wikipedia.org/wiki/Document-oriented_database for more information.
 //
+// Subpackages contain distinct implementations ("drivers") of docstore for public
+// cloud services like Google Cloud Firestore and Amazon DynamoDB as well as any
+// MongoDB service hosted by a cloud provider or on-premise. For example, you can
+// use Azure Cosmos DB and Amazon DocumentDB via the MongoDB driver. There is also
+// memdocstore, an in-process, in-memory implementation suitable for testing and
+// development.
+//
 // In docstore, documents are grouped into collections, and each document has a key
-// that is unique in its collection. You can add, delete, modify and retrieve
+// that is unique in its collection. You can add, retrieve, modify and delete
 // documents by key, and you can query a collection to retrieve documents that match
 // certain criteria.
 //
-// Subpackages contain distinct implementations ("drivers") of docstore for various
-// providers, including MongoDB, Google Cloud Firestore, and Amazon DynamoDB.
-// You can use Azure Cosmos DB and Amazon DocumentDB via the MongoDB driver.
-// There is also memdocstore, an in-process, in-memory implementation suitable
-// for testing and development.
-//
-// Your application should import one of these provider-specific subpackages and use
+// Your application should import one of the service-specific subpackages and use
 // its exported functions to create a *Collection; do not use the NewCollection
 // function in this package. For example:
 //
 //   coll, err := memdocstore.OpenCollection("SSN", nil)
 //   if err != nil {
-//     return fmt.Errorf("opening collection: %v", err)
+//       return fmt.Errorf("opening collection: %v", err)
 //   }
 //   defer coll.Close()
 //   // coll is a *docstore.Collection
 //
 // Then, write your application code using the *Collection type. You can easily
-// reconfigure your initialization code to choose a different provider.
+// reconfigure your initialization code to choose a different driver.
 // You can develop your application locally using memdocstore, or deploy it to
 // multiple Cloud providers. You may find http://github.com/google/wire useful
 // for managing your initialization code.
 //
-// Alternatively, you can construct a *Collection via a URL and the OpenCollection
-// function in this package. See https://gocloud.dev/concepts/urls for more
+// Alternatively, you can construct a *Collection via a URL and the
+// docstore.OpenCollection function. See https://gocloud.dev/concepts/urls for more
```

Comment:
Do we need to add `docstore` here since it is in this package. When the user looks at the godoc, this would be in the same page.

Reply:
Done